src/lib/free-queue/test/free-queue.test.js

```diff
@@ -0,0 +1,162 @@
+    });
+
+    it('should allocate the correct amount of memory', () => {
+      const dataByteSize = channelCount * bufferLength * Float32Array.BYTES_PER_ELEMENT;
+      expect(freeQueue.getPointer()).to.be.instanceof(ArrayBuffer);
+      expect(freeQueue.getPointer().byteLength).to.equal(dataByteSize);
+    });
+  });
+
+  describe('Channel Adaptation', () => {
+    it('should adapt to a new channel count within limits', () => {
+      freeQueue.adaptChannel(3);
+      expect(freeQueue.numberOfChannels).to.equal(3);
+    });
+  });
+
+  describe('Pushing and Pulling Data', () => {
+    it('should correctly push and pull data', () => {
+      const testData = [new Float32Array(bufferLength).fill(1), new Float32Array(bufferLength).fill(2)];
+      freeQueue.push(testData);
+
+      const outputData = [new Float32Array(bufferLength), new Float32Array(bufferLength)];
+      freeQueue.pull(outputData);
+
+      expect(outputData[0]).to.deep.equal(testData[0]);
+      expect(outputData[1]).to.deep.equal(testData[1]);
+    });
+
+    it('should handle buffer overflow correctly', () => {
+      const testData = [new Float32Array(bufferLength * 2).fill(1), new Float32Array(bufferLength * 2).fill(2)];
+      freeQueue.push(testData);
+
+      expect(freeQueue.framesAvailable).to.equal(bufferLength);
+    });
+
+    it('should not pull data when buffer is empty', () => {
+      const outputData = [new Float32Array(bufferLength), new Float32Array(bufferLength)];
+      freeQueue.pull(outputData);
+
+      expect(outputData[0]).to.deep.equal(new Float32Array(bufferLength));
+      expect(outputData[1]).to.deep.equal(new Float32Array(bufferLength));
+    });
+
+    it('should manage partial data pulls', () => {
+      const testData = [new Float32Array(bufferLength).fill(1), new Float32Array(bufferLength).fill(2)];
+      freeQueue.push(testData);
+
+      const partialOutput = [new Float32Array(bufferLength / 2), new Float32Array(bufferLength / 2)];
+      freeQueue.pull(partialOutput);
+
+      expect(partialOutput[0]).to.deep.equal(new Float32Array(bufferLength / 2).fill(1));
+      expect(partialOutput[1]).to.deep.equal(new Float32Array(bufferLength / 2).fill(2));
+      expect(freeQueue.framesAvailable).to.equal(bufferLength / 2);
+    });
+
+    it('should handle multiple push and pull cycles', () => {
+      const testData = [new Float32Array(bufferLength).fill(1), new Float32Array(bufferLength).fill(2)];
+      
+      for (let i = 0; i < 5; i++) {
+        freeQueue.push(testData);
+
+        const outputData = [new Float32Array(bufferLength), new Float32Array(bufferLength)];
+        freeQueue.pull(outputData);
+
+        expect(outputData[0]).to.deep.equal(testData[0]);
+        expect(outputData[1]).to.deep.equal(testData[1]);
+        expect(freeQueue.framesAvailable).to.equal(0);
+      }
+    });
+  });
+
+  describe('Error Handling', () => {
+    it('should return null for invalid channel index in getChannelData', () => {
```

Comment:
Can we port these tests back to the FreeQueueSAB suite too?

Reply:
This is possible to do. I'll do this in a new PR after merging this.